cost-analyzer/templates/kubecost-metrics-service-template.yaml

```diff
@@ -6,10 +6,17 @@ kind: Service
 metadata:
   name: {{ template "kubecost.kubeMetricsName" . }}
   labels:
-    {{ include "cost-analyzer.commonLabels" . | nindent 4 }}
+{{ include "cost-analyzer.commonLabels" . | nindent 4 }}
+{{- if (or .Values.kubecostMetrics.exporter.service.annotations .Values.kubecostMetrics.exporter.prometheusScrape) }}
   annotations:
+{{- if .Values.kubecostMetrics.exporter.service.annotations }}
+{{ toYaml .Values.kubecostMetrics.exporter.service.annotations | indent 4 }}
+{{- end }}
+{{- if .Values.kubecostMetrics.exporter.prometheusScrape }}
```

Comment:
This feels like it should have a "kindis" check followed up by a default if not exists to ensure that existing users don't need to modify their values.yaml to make this work. I know that the default is set lower in values.yaml but not all users modify values before upgrading, in which case these labels will be dropped silently.

https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/templates/cost-analyzer-deployment-template.yaml#L537